### PR TITLE
ct bdd2d added

### PR DESCRIPTION
**Pull request by SoniaMinseoKim**
needed packages "xitorch"

### CODE REVIEW

**Comment by guanhuaw** [comment]
Hi Sonia,

Thanks for the excellent work! I learned much from reading your codes. Here are some very minor suggestions:

For `map2x`, `map2y`, and `intergrate1D`, could you please add some comments on their functions? Also, for `Interp1D`, can we use PyTorch's native implementation, if possible? Or we may want to declare in the project configuration (`setup.py`) to install `xitorch`.

For bdd class, could you please add comments following the Gmri case? The advantage is that it will be automatically compiled into the [tutorial](https://mirtorch.readthedocs.io/en/latest/). Also, it would be great to annotate the default values and type similar to Gmri case. Plus, we can add it to `__init__.py`. If `size_in` and `size_out` are already defined by other variables, we do not necessarily need to include them in the initialization function.

Also looking forward to your jupyter notebook as a tutorial to the community!

Best,
Guanhua

**Comment by ZongyuLi-umich** [comment]
Sonia is using the xitorch.Interp1D because the native pytorch implementation does not support extrapolation. However, she can first do [replicate padding](https://pytorch.org/docs/stable/generated/torch.nn.ReplicationPad1d.html) and then interpolate.

**Comment by guanhuaw** [comment]
Thanks, Zongyu, for the clarification. If we need more functions of `xitorch` in other CT/SPECT cases, we can add it to dependence - `xitorch` does not require many other packages. Or we may implement a simple native implementation, as suggested. What do you think?

**Comment by ZongyuLi-umich** [comment]
> Thanks, Zongyu, for the clarification. If we need more functions of `xitorch` in other CT/SPECT cases, we can add it to dependence - `xitorch` does not require many other packages. Or we may implement a simple native implementation, as suggested. What do you think?

Yes, I agree with you.

**Comment by SoniaMinseoKim** [comment]
Hello! I have added all the necessary comments, please let me know if there is anything else I must change!

Also, for the `xitorch`, presently I have added the package to `setup.py` for now but I tried to implement the function using replicate padding and interpolate functions from native PyTorch but I am encountering some issues with the precision of the output values (refer to the code below):

`def inter1d2(idx, val, query):
    # Sort idx and val tensors in ascending order based on idx
    sorted_indices = torch.argsort(idx)
    sorted_idx = idx[sorted_indices]
    sorted_val = val[sorted_indices]
    
    sorted_query = torch.argsort(query)
    sorted_q = query[sorted_query]
    
    # Pad the sorted_val tensor with replication padding
    if sorted_idx[0] > sorted_q[0]:
        left = sorted_idx[0] - sorted_q[0]
        left = left.type(torch.int64)
    else: 
        left = 0
    
    if sorted_idx[-1] < sorted_q[-1]:
        right = sorted_q[-1] - sorted_idx[-1]
        right = right.type(torch.int64)
    else:
        right = 0
    
    padded_val = torch.nn.ReplicationPad1d((left, right))(sorted_val.unsqueeze(0))

    interpolated_val = torch.nn.functional.interpolate(padded_val.unsqueeze(0), size=query.size(), mode='linear')

    Pdk = interpolated_val.squeeze(0).squeeze(0)

    return Pdk`

When tested with the example tensors:
`torch.tensor([0, 1, 2], dtype=torch.float64), 
  torch.tensor([3, 2, 1], dtype=torch.float64), 
  torch.tensor([-2, 0.5, 1.5, 3], dtype=torch.float64)`

The output gives:
`tensor([3.0000, 3.0000, 1.7500, 1.0000], dtype=torch.float64)`
compared to the output of xitorch:
`tensor([3.0000, 2.5000, 1.5000, 1.0000], dtype=torch.float64)`

Is there something wrong with my implementation? 

Also, I will add the Jupyter notebook soon!

**Comment by ZongyuLi-umich** [comment]
Sonia, thanks for the work! One possible solution is to use [grid_sample](https://pytorch.org/docs/0.3.0/nn.html#torch.nn.functional.grid_sample) for the interpolation.

**Comment by SoniaMinseoKim** [comment]
Hi Guanhua, it turns out that the implementation is not trivial and I also found a GitHub page that implemented linear 1d interpolation in Pytorch but it required another package. Overall, I think we can just call `xitorch`. What do you think?

**Comment by guanhuaw** [comment]
No worries! I will add `xitorch` to the dependence. Are you using the latest commit of `xitorch`?

**Comment by SoniaMinseoKim** [comment]
Yes, that is correct!
